anesthetic/samples.py

```diff
@@ -70,7 +76,8 @@ def __init__(self, *args, **kwargs):
                                  "instead which has the same features as "
                                  "MCMCSamples and more. MCMCSamples should be "
                                  "used for MCMC chains only." % root)
-            w, logL, samples = reader.samples()
+            discard_burnin = kwargs.pop('discard_burnin', False)
+            w, logL, samples = reader.samples(discard_burnin=discard_burnin)
```

Comment:
Is there any way one could accidentally pass discard_burnin from a NestedSamples object? Given that this is something that people can do if they are not used to working with nested sampling chains, if it is possible we should probably put in a guard against this that throws a ValueError if discard_burnin is passed to NestedSamples.

Reply:
A few lines above this, there already is a guard against using `MCMCSamples` on nested sampling data, so no danger on that front.

When passing to `NestedSamples` it won't reach where `discard_burnin`/`burn_in` is beeing pop-ed, as that happens in the `if root is not None` block of `MCMCSamples` which doesn't get executed by `NestedSamples`.